src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -112,13 +112,15 @@ protected function makeSureFieldHasName($field)
         }
 
         if (is_string($field)) {
-            return ['name' => $field];
+            return ['name' => $this->trimWhitespacesFromName($field)];
         }
 
         if (is_array($field) && ! isset($field['name'])) {
             abort(500, 'All fields must have their name defined');
         }
 
+        $field['name'] = $this->trimWhitespacesFromName($field['name']);
```

Comment:
```suggestion
        $field['name'] = Str::squish($field['name']);
```

Reply:
Same note, alternative would be 
```suggestion
        $field['name'] = Str::replace(' ', '', $field['name']);
```

Reply:
What if the left two spaces? `permission,  roles` ? `Str::replace()` would do the same as squish here, remove the extra space only.